### PR TITLE
MVP

### LINKED ISSUE #10: Implement reusable layout for Main header view
image

### LINKED ISSUE #11: Implement Transaction List redesign
- [ ] Transaction List
- [ ] Value Transfer Transaction
- [ ] Data Request Transaction

### LINKED ISSUE #12: Implement Addresses redesign
- [ ] Address list
- [ ] Generate address flow

### LINKED ISSUE #13: Implement Create VTT flow redesign
- Create VTT Form step
- Review VTT step

### LINKED ISSUE #15: Implement Settings redesign
#### Export xprv:
- [ ] Fix import wallet from xprv
- [ ] Add export xprv feature
#### General settings:
- [ ] Add select unit

### LINKED ISSUE #28: Fix mobile size responsive
- [ ] Fix mobile margins
- [ ] Fix main layout to fit mobile height (avoid scroll in welcome view)
- [ ] Fix transaction info in transaction list to fit mobile width

### PR DESCRIPTION
**Pull request by gabaldon**
close [#255](https://github.com/witnet/internal/issues/255)
close #11 
close #12 
close #10 
close #13 
close #15
close #28

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.witnet.witnet_wallet">
    <application
+        :name="${applicationName}"
```
Comment (Tommytrg) [inline android/app/src/main/AndroidManifest.xml:4]:
```suggestion
        android:name="${applicationName}"
```

**Review by Tommytrg — COMMENTED** [review]
Could you remove this file: [flutter_01.png](https://github.com/witnet/witnet_wallet/pull/25/files#diff-377e084fd300c1cfa520004b9abfc4a8dc16426d773b68caa733e96180fac611)?

File: node_modules/.yarn-integrity [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (Tommytrg) [inline node_modules/.yarn-integrity:1]: Remove vile

File: pubspec.yaml [hunk 3]
```diff
@@ -8,50 +8,62 @@ version: 0.0.1-dev
...
-
   bloc: ^8.0.3
   get_it: ^7.2.0
   flutter_bloc: ^8.0.1
   flutter_spinkit: ^5.0.0
   shared_preferences: ^2.0.7
   carousel_slider: ^4.1.1
   path_provider: ^2.0.1
-  flutter_svg: ^0.22.0
+  #flutter_svg: ^0.22.0
```
Comment (Tommytrg) [inline pubspec.yaml:31]:
```suggestion
```

File: pubspec.yaml [hunk 4]
```diff
@@ -8,50 +8,62 @@ version: 0.0.1-dev
...
-
+  toml: ^0.13.1
+  decimal: 2.3.2
 
-  witnet: 0.1.4
-  #witnet:
-  #  path: ../witnet
+  # witnet: 0.1.4
+  witnet:
+    path: ../witnet.dart
```
Comment (Tommytrg) [inline pubspec.yaml:50]: Could we add the right version?

File: pubspec.yaml [hunk 5]
```diff
@@ -8,50 +8,62 @@ version: 0.0.1-dev
...
   table_calendar: ^3.0.3
   qr_flutter: ^4.0.0
   url_launcher: ^6.0.10
   flutter_dotenv: ^5.0.2
   date_format: ^2.0.5
   qr_code_scanner: ^1.0.1
   formz: ^0.4.1
+  flutter_hooks: ^0.18.5+1
+
+
```
Comment (Tommytrg) [inline pubspec.yaml:60]:
```suggestion
```

File: test/keychain_test.dart [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+void main() async {
+  test('Password Encryption Test', () async {
+    KeyChain keyChain = KeyChain();
+    final String password = 'Witnet';
+    final String incorrectPassword = 'Witnet1';
+
+    var encoded = bytesToHex(keyChain.encode(password));
+
+    ///
```
Comment (Tommytrg) [inline test/keychain_test.dart:13]:
```suggestion
```

File: test/create_vtt_form_test.dart [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
```
Comment (Tommytrg) [inline test/create_vtt_form_test.dart:1]: This test is empty. Could we remove it?

File: lib/bloc/explorer/explorer_bloc.dart [hunk 8]
```diff
@@ -81,161 +82,205 @@ class ExplorerBloc extends Bloc<ExplorerEvent, ExplorerState> {
...
-      /// if the gap limit is not maintained then generate additional accounts
-      ///
-      int lastExternalIndex = dbWallet.externalAccounts.length;
-      while (externalGap <= EXTERNAL_GAP_LIMIT) {
-        Account account = await dbWallet.generateKey(
-            index: lastExternalIndex, keyType: KeyType.external);
-        lastExternalIndex += 1;
+  Future<void> syncWalletRoutine(Wallet dbWallet) async {
+    /// verify gap limit
+    /// external chain
```
Comment (Tommytrg) [inline lib/bloc/explorer/explorer_bloc.dart:97]: What these comment refer to?

File: lib/bloc/transactions/value_transfer/vtt_status/vtt_status_bloc.dart [hunk 9]
```diff
@@ -33,15 +31,14 @@ class BlocStatusVtt extends Bloc<VttStatusEvent, VttStatusState> {
   static VttStatusState get initialState => UnknownHashState();
 
   Future<void> checkStatus(String transactionHash) async {
-    ApiExplorer apiExplorer = Locator.instance.get<ApiExplorer>();
+    // ApiExplorer apiExplorer = Locator.instance.get<ApiExplorer>();
```
Comment (Tommytrg) [inline lib/bloc/transactions/value_transfer/vtt_status/vtt_status_bloc.dart:34]: Are we calling checkStatus? If not, could we remove the method or add a comment explaining why we should keep it commented?

File: lib/screens/dashboard/api_dashboard.dart [hunk 10]
```diff
@@ -1,14 +1,30 @@
...
-import 'package:witnet_wallet/util/storage/database/db_wallet.dart';
+import 'package:witnet_wallet/util/storage/database/wallet.dart';
+
+import 'package:witnet_wallet/util/storage/database/wallet_storage.dart';
+import 'package:witnet_wallet/widgets/address.dart';
 
 class ApiDashboard {
-  DbWallet? _dbWallet;
+  WalletStorage? _walletStorage;
+  //TODO: define wallet type
```
Comment (Tommytrg) [inline lib/screens/dashboard/api_dashboard.dart:9]: What are the different types of wallets?

File: lib/screens/dashboard/bloc/dashboard_bloc.dart [hunk 11]
```diff
@@ -2,67 +2,93 @@ import 'package:bloc/bloc.dart';
...
 class DashboardBloc extends Bloc<DashboardEvent, DashboardState> {
+  // dbWallet: DbWallet(
+  //               externalXpub: null,
+  //               internalXpub: null,
+  //               xprv: null,
+  //               walletName: null,
+  //               lastSynced: -1,
+  //               walletDescription: '',
+  //               externalAccounts: {},
+  //               internalAccounts: {}),
```
Comment (Tommytrg) [inline lib/screens/dashboard/bloc/dashboard_bloc.dart:37]:
```suggestion
```

File: lib/util/storage/database/database_service.dart [hunk 12]
```diff
@@ -32,215 +38,211 @@ class DBException {
...
+        bool valid = await keyChain.validatePassword(key, password);
+        if (valid) {
+          unlocked = true;
+        }
+        return valid;
+      }
+    } catch (e) {
+      return false;
     }
+    return false;
```
Comment (Tommytrg) [inline lib/util/storage/database/database_service.dart:179]: IMHO this is easier to read. What do you think?
```suggestion
        if (!keyExists) {
          return false
        }
        
        String? key = await keyChain.getKey(_database);
  
        bool valid = await keyChain.validatePassword(key, password);
        if (valid) {
          unlocked = true;
        }
        return valid;
      }
    } catch (e) {
      return false;
    }

```

File: lib/util/storage/database/encrypt/keychain.dart [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+    if (exists) {
+      String key = await getKey(databaseClient);
+
+      bool valid = await validatePassword(key, oldPassword!);
+
+      if (valid) {
+        await _store.record('keychain').add(databaseClient, encodedKey);
+      } else {
+        return false;
+      }
```
Comment (Tommytrg) [inline lib/util/storage/database/encrypt/keychain.dart:104]:
```suggestion
      if (!valid) {
        return false;
      }
      await _store.record('keychain').add(databaseClient, encodedKey);
```

File: lib/util/storage/database/wallet_storage.dart [hunk 14]
```diff
@@ -0,0 +1,38 @@
+import 'package:witnet/data_structures.dart';
+import 'package:witnet_wallet/util/storage/database/wallet.dart';
+
+import 'balance_info.dart';
+
+/// DbWallet formats the wallet for the database
+///
```
Comment (Tommytrg) [inline lib/util/storage/database/wallet_storage.dart:7]:
```suggestion
```

File: lib/widgets/layouts/dashboard_layout.dart [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  /// _goToSettings
```
Comment (Tommytrg) [inline lib/widgets/layouts/dashboard_layout.dart:49]:
```suggestion
```

**Comment by gabaldon** [comment]
duplicated in #36

**Pull request closed without merging**